readthedocs/doc_builder/python_environments.py

```diff
@@ -87,6 +98,70 @@ def venv_bin(self, filename=None):
             parts.append(filename)
         return os.path.join(*parts)
 
+    def environment_json_path(self):
+        """Return the path to the ``environment.json`` file for this venv."""
+        return os.path.join(
+            self.venv_path(),
+            'environment.json',
+        )
+
+    @property
+    def is_obsolete(self):
```

Comment:
I wrote this at PythonEnvironment level, but @ericholscher said that conda won't be affected by this. Should I leave this here or move it to Virtualenv and write something customized by Conda?

Reply:
Honestly not sure. @willingc do you know anything about how conda will be effected by underlying docker image changes? My thought if that it's self-contained, but not really sure.

Reply:
Hmm... not sure. My understanding is docker images are built in layers so if you change an underlying layer it impacts the layers above.

Reply:
This would be changing out the underlying system python or conda version, I believe, is that main worry. We have the files on disk from a previous run, but we're hoping to let users change their docker image. We're planning to blow away the virtualenvs, because it breaks, but curious if conda will be effected.

Reply:
@willingc if a virtualenv is created with python 2.7.13 and we change the python version to 2.7.14, the venv stop working and need to be re-created. In the case of Conda we are not sure about that case but also the case that the conda environment was created with conda 4.x and then it's ran with conda 4.(x+1). What would happen in this case? Do you know?

ccing @juanlu001 who was my Conda teacher :)